Update cosigned docs with new ClusterImagePolicy usage

DennyHoang opened the pull request:
#### Summary
Add supporting documentation on ClusterImagePolicy configuration and usage.

#### Ticket Link
Addressing https://github.com/sigstore/sigstore-website/issues/139

#### Release Note
```release-note
* Added documentation for ClusterImagePolicy configuration and usage
```

vaikas commented on content/en/cosign/kubernetes.md:
```diff
@@ -45,11 +45,174 @@ The `cosigned` admission controller can be used to enforce policy on a Kubernete
+
+### Admission of Images
+
+An image is admitted after it has been validated against all `ClusterImagePolicy` that matches and
```
Maybe add a link to 'configuring image patterns' below?

vaikas commented on content/en/cosign/kubernetes.md:
```diff
@@ -45,11 +45,174 @@ The `cosigned` admission controller can be used to enforce policy on a Kubernete
+
+#### Configuring `key` Authorities
+
+When a policy is selected to be evaluated against for the matched image, the authorities will be used to validate signatures.
```
s/against for/against/

vaikas commented on content/en/cosign/kubernetes.md:
```diff
@@ -45,11 +45,174 @@ The `cosigned` admission controller can be used to enforce policy on a Kubernete
+#### Configuring `key` Authorities
+
+When a policy is selected to be evaluated against for the matched image, the authorities will be used to validate signatures.
+If any one authority is satisfied and a signature is validated, the policy is validated.
```
maybe it's me that 'any one' reads funny, maybe at least one?

vaikas commented on content/en/cosign/kubernetes.md:
```diff
@@ -45,11 +45,174 @@ The `cosigned` admission controller can be used to enforce policy on a Kubernete
+1. A valid signature was obtained for `policy1` with any of the `policy1` authorities
+1. No valid signature was obtained for `policy2` with any of the `policy2` authorities
+1. The image is not admitted
+
```
We should somewhere mention what happens _iff_ an image does not match any policy. It means in current implementation it will fall through to the "old" behaviour, which can result in something like this:
https://github.com/sigstore/cosign/issues/1653

vaikas commented on content/en/cosign/kubernetes.md:
```diff
@@ -45,11 +45,174 @@ The `cosigned` admission controller can be used to enforce policy on a Kubernete
 Enforcement is configured on a per-namespace basis, and multiple keys are supported.
 
 We're actively working on more features here, including support for Attestations.
+
```
This is really for L53, but do we anywhere (I may have missed) mention how you control, aka, what label (and an example on how to enable the validation?)

something like, to enable validation on a namespace `demo`, you would do:
```
kubectl label namespace demo cosigned.sigstore.dev/include=true
```

DennyHoang replied:
It is mentioned in the `installation` doc under the `cosigned`. But no harm replicating that detail in this doc.

Otherwise, moving that detail to this file and then in the installation doc, I could link it back to this file for that section.

vaikas commented on content/en/cosign/kubernetes.md:
```diff
@@ -45,11 +45,174 @@ The `cosigned` admission controller can be used to enforce policy on a Kubernete
+
+TLog specifies the URL to a transparency log that holds signature and public key information.
+
+When `tlog` key is specified with no value `tlog: {}`, then the public rekor instance will be used.
```
I think you can just leave it out, as in not specify anything at all?

vaikas replied:
So, maybe:
When `tlog` is not specified, then the public rekor instance will be used.

vaikas reviewed (commented):
Thanks so much!!!

DennyHoang commented:
Addressed PR comments:
- Fixed grammar
- Added `Depcreated Cosigned Verification Behavior` Section
- Moved `Enable Cosigned Admission Controller for Namespaces` section to `kubernetes.md` and have it linked from `installation.md` instead
- added "no policy matched" scenario to section `Admission of Images`
- Mentioned currently only supporting ECDSA keys

cc: @vaikas

vaikas reviewed (approved):
Awesome! Thanks for doing this! I'll take a crack at adding documentation for attestations. I'm at an offsite this week though, so will be a bit slower than usual.

hectorj2f reviewed (changes requested):
@DennyHoang Thanks :).
Could you add the last properties added to the CIP e.g. policy ...?

DennyHoang commented:
@hectorj2f I have not gotten familiar with the new CIP policy attestation fields yet. So I would prefer adding that documentation in a different PR.
